PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -3968,9 +3968,12 @@ void ScintillaEditView::hideLines()
 	size_t startMarker = startLine - 1;
 	size_t endMarker = endLine + 1;
 
-	// Remove all previous markers in between new ones
+	// Previous markers must be removed in the selected region:
+
+	removeMarker(startMarker, 1 << MARK_HIDELINESBEGIN);
 	for (size_t i = startLine; i <= endLine; ++i)
 		removeMarker(i, (1 << MARK_HIDELINESBEGIN) | (1 << MARK_HIDELINESEND));
+	removeMarker(endMarker, 1 << MARK_HIDELINESEND);

```

Comment:
@donho
Thank you for the feedback!
The reason I didn't combine 3 `removeMarker()` calls into a single `for` loop is because it would slightly change the way lines are hidden — not better or worse, just different. Let's say we have the following situation:

![0](https://github.com/user-attachments/assets/1504c73a-af3f-41e7-9472-20a95b902b25)

The question is: what should happen if we try to hide line number 7 (just that line)? Currently, and after my PR, the program hides just line number 7:

![1](https://github.com/user-attachments/assets/81c74c62-7264-4e79-b08e-bfba744e2dbc)

However, after following your suggestion, the program will behave like this:

![2](https://github.com/user-attachments/assets/1c83e689-98ad-48bb-abae-e6ae7927c467)

Instead of overlapping begin and end markers, it will also merge the hidden section and hide line number 6. In other words, this suggestion would undo #13681.

Why this happens?

![00](https://github.com/user-attachments/assets/c6612144-3110-4b30-acfd-e0ffc163750f)

Hiding line number 7 will not only try to remove the **non-existent begin** marker on line number 6, but also the **existing end** marker on the same line and this behavior will propagate further upwards, causing the merge effect.

Please let me know what you think, and I'll align myself with your opinion.

Reply:
Yes, indeed.
After playing your crash example, I realize the difference.
OK, let's keep the current code.